### PR TITLE
OD-339 [Fix] It's impossible to save LFD with an LFD filter option

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 
## Issue
OD-339 https://weboo.atlassian.net/browse/OD-339
## Description
It's impossible to save LFD with an LFD filter option which uses "Is empty" or "Is not empty" filter logics
## Screenshots/screencasts
http://somup.com/crVO2sogZd
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko @ivandevupp

### CODE REVIEW

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Does not conform to coding standards. Does not work in IE 11.

**Comment by AndrRyaz** [comment]
@tonytlwu i changed the logic of the condition to `indexOf `

File: js/interface-lists.js [hunk 1]
```diff
@@ -457,8 +457,20 @@ function attahObservers() {
         if (widgetData.filterOptions.length) {
           var filterError = [];
           var filterFieldValues = [];
+          var nonValidatedLogicOptions = [
```
Comment (tonytlwu) [inline js/interface-lists.js:460]: The variable name doesn't make sense.

1. It's `invalid`, not `nonValid`.
2. I don't understand why these are invalid. They look like valid logic options to me. This shows that the variable name is inappropriately named.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/interface-lists.js (hunk 1 above)
Comment (AndrRyaz) [reply js/interface-lists.js:460]: @tonytlwu let me tell you what I mean here please, the logic of this check is to exclude checking of the Value type field when it is hidden and inactive, this array is responsible for each type of logic in which this field is hidden

Comment (tonytlwu) [reply js/interface-lists.js:460]:
```suggestion
          var logicOptionsWithoutValues = [
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Comment by AndrRyaz** [comment]
@tonytlwu, resolved

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: Medium
Affected: Studio users configuring LFD filter options and conditions
Verify: Filter options can be correctly saved and loaded

Squash and merge: Ignore logic options that don't need values

**Pull request merged**